### DISCUSSION TITLE
How would Rhai perform on a Cortex M4?

### DISCUSSION THREAD

**Discussion by paulrouget** [rhaiscript/rhai#462] [General]
I'm thinking of using Rhai on a nRF52. A Rhai function could be called up to ~30 times a second (twice a second on average). I've been reading the Minimal Build section of the book, and it feels like it could do the job.

Any feedback on running Rhai on such hardware?

**Comment by schungx** [comment]
Not very sure of it, but you can always try!  You're encouraged to contribute your benchmark results and I'll put them in the docs!

Rhai is pretty optimized, so it doesn't run too slowly, but it is an AST-walker, so it would be slower than a bytecodes interpreter.

Furthermore, when you use minimal builds, the compiler is likely not going to inline a lot of function calls, which may make things slower.

But the Cortex M4 is a pretty high-performance architecture, so I think you shouldn't get any problems running 30 fps unless your scripts are very very complex.

You should turn on `unchecked` and turn off unnecessary features to maximize your speed.  Also, use `only_i32` would probably save you data size.  In this respect, follow the performance build recommendations.

**Comment by schungx** [comment]
For discussions, I also suggest you join our Discord channel.

**Comment by schungx** [comment]
Closing this for now. Feel free to reopen if necessary.
